### PR TITLE
[DXP-994] Shorten command line errors when user passes invalid params

### PR DESCRIPTION
**Pull request by stan-r-ds**

### CODE REVIEW

File: core/src/main/java/dev/streamx/cli/ExecutionExceptionHandler.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import picocli.CommandLine.ParseResult;
 
 @ApplicationScoped
-public class ExceptionHandler implements IExecutionExceptionHandler {
+public class ExecutionExceptionHandler implements IExecutionExceptionHandler {
```
Comment (stan-r-ds) [inline core/src/main/java/dev/streamx/cli/ExecutionExceptionHandler.java:12]: Added ParameterExceptionHandler, so make this existing handler's name more meaningful

File: core/src/main/java/dev/streamx/cli/ParameterExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  public int handleParseException(ParameterException ex, String[] args) {
+    log.error("Parameter exception occurred.", ex);
+
+    CommandLine cmd = ex.getCommandLine();
+    PrintWriter writer = cmd.getErr();
+
+    writer.println(ex.getMessage());
+    UnmatchedArgumentException.printSuggestions(ex, writer);
+    writer.print(cmd.getHelp().fullSynopsis());
+
```
Comment (stan-r-ds) [inline core/src/main/java/dev/streamx/cli/ParameterExceptionHandler.java:29]: In general, this is the pr

File: core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java [hunk 3]
```diff
@@ -37,14 +37,7 @@ public void shouldRejectChannellessIngestion(QuarkusMainLauncher launcher) {
...
                                  (<channel> <key>)
-        Send unpublication trigger
-              <channel>   Channel that message will be published to
-              <key>       Message key
-          -h, --help      Show this help message and exit.
-              --ingestion-url=<propagateIngestionUrl>
-                          Address of 'rest-ingestion-service'
-                            Default: http://localhost:8080
-          -V, --version   Print version information and exit.""");
+        Try 'streamx unpublish --help' for more information.""");
```
Comment (stan-r-ds) [inline core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java:40]: Here we have it, shorter command line boilerplate, especially in the below test for publish command

Comment (stan-r-ds) [reply core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java:40]: I also verified, that the only remaining long command line errors, are:
1. `shouldRejectInvalidFile`
```
          Payload could not be parsed.
                    
          Supplied payload:
          asdf{][
                    
          Make sure that:
           * it's valid JSON,
           * object property names are properly single-quoted (') or double-quoted ("),
           * strings are properly single-quoted (') or double-quoted (")
                    
          Details: Unrecognized token 'asdf': was expecting (JSON String, Number, Array, Object
           or token 'null', 'true' or 'false')
           at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled);
           line: 1, column: 6]
```

2. `shouldRejectInvalidDataJson`
```
Payload could not be parsed.
                    
          Supplied payload:
          asdf{][
                    
          Make sure that:
           * it's valid JSON,
           * object property names are properly single-quoted (') or double-quoted ("),
           * strings are properly single-quoted (') or double-quoted (")
                    
          Details: Unrecognized token 'asdf': was expecting (JSON String, Number, Array, Object
           or token 'null', 'true' or 'false')
           at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled);
           line: 1, column: 6]
```

Those seem OK to remain long/descriptive

File: core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java [hunk 4]
```diff
@@ -58,50 +51,7 @@ public void shouldRejectIngestionWithMissingParameter(QuarkusMainLauncher launch
...
-            * if the value is a JSON fragment, use -j option
-            * if the value is a binary value fragment use -b option
-                
-          -b, --binary-fragment=<binary>
-                              Defines a payload fragment of binary type
-          -j, --json-fragment=<json>
-                              Defines a payload fragment of JSON node type
-          -s, --string-fragment=<string>
-                              Defines a payload fragment of string type""");
+        Try 'streamx publish --help' for more information.""");
```
Comment (jgardo-ds) [inline core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java:54]: If it's possible - add newline to separate suggestion "Try" from rest of information.

File: core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java [hunk 5]
```diff
@@ -114,14 +64,7 @@ public void shouldRejectKeylessIngestion(QuarkusMainLauncher launcher) {
...
                                  (<channel> <key>)
-        Send unpublication trigger
-              <channel>   Channel that message will be published to
-              <key>       Message key
-          -h, --help      Show this help message and exit.
-              --ingestion-url=<propagateIngestionUrl>
-                          Address of 'rest-ingestion-service'
-                            Default: http://localhost:8080
-          -V, --version   Print version information and exit.""");
+        Try 'streamx unpublish --help' for more information.""");
```
Comment (jgardo-ds) [inline core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java:67]: If it's possible - add newline to separate suggestion "Try" from rest of information.

File: core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java (hunk 3 above)
Comment (jgardo-ds) [inline core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java:40]: If it's possible - add newline to separate suggestion "Try" from rest of information.

File: core/src/main/java/dev/streamx/cli/ParameterExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    CommandSpec spec = cmd.getCommandSpec();
+    writer.printf("Try '%s --help' for more information.%n", spec.qualifiedName());
```
Comment (jgardo-ds) [inline core/src/main/java/dev/streamx/cli/ParameterExceptionHandler.java:31]: If it's possible - add newline to separate suggestion "Try" from rest of information.

**Review by jgardo-ds — APPROVED** [review]

File: core/src/main/java/dev/streamx/cli/ParameterExceptionHandler.java (hunk 6 above)
Comment (stan-r-ds) [reply core/src/main/java/dev/streamx/cli/ParameterExceptionHandler.java:31]: Done

File: core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java (hunk 5 above)
Comment (stan-r-ds) [reply core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java:70]: Done

File: core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java (hunk 4 above)
Comment (stan-r-ds) [reply core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java:56]: Done

File: core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java (hunk 3 above)
Comment (stan-r-ds) [reply core/src/test/java/dev/streamx/cli/ingestion/IngestionArgumentsValidationTest.java:41]: Done

**Pull request merged**